### PR TITLE
Add lowering to special.modified_bessel_i0

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119187
* __->__ #118993

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -787,6 +787,15 @@ def bessel_j1(x):
     return make_pointwise(fn)(x)
 
 
+@register_lowering(
+    [aten.special_modified_bessel_i0],
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+)
+def modified_bessel_i0(x):
+    fn = ops_wrapper("modified_bessel_i0")
+    return make_pointwise(fn)(x)
```
Comment (peterbell10) [inline torch/_inductor/lowering.py:796]: BTW for simple pointwise lowerings like this you can use the `register_pointwise` helper
```suggestion
register_pointwise(
    aten.special_modified_bessel_i0,
    name="modified_bessel_i0",
    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
)
```

And move the definition down by the other `register_pointwise` calls.

**Review by peterbell10 — APPROVED** [review]

File: torch/_inductor/lowering.py (hunk 1 above)
Comment (pearu) [reply torch/_inductor/lowering.py]: Thanks for the hint! I ended up using
```python
register_pointwise_numeric(aten.special_modified_bessel_i0)
```
with prefix `special_` removed within `register_pointwise`.

File: torch/_inductor/lowering.py [hunk 2]
```diff
@@ -596,7 +596,7 @@ def register_pointwise(
     use_libdevice_for_f64=False,
 ):
     """A pointwise function that maps ops.{name} to inputs"""
-    name = name or aten_fn.__name__
+    name = name or aten_fn.__name__.removeprefix('special_')
```
Comment (peterbell10) [inline torch/_inductor/lowering.py:599]: I would prefer passing `name` explicitly rather than having special cases for special in this function.

Comment (pearu) [reply torch/_inductor/lowering.py:599]: Understood. Notice that there are more than 50 special functions. How about introducing `register_pointwise_special` helper function that removes the prefix with the assert that the prefix `special_` must exists. What do you think?

Comment (pearu) [reply torch/_inductor/lowering.py:599]: Or have:
```python
register_pointwise_numeric(aten.special_modified_bessel_i0, name='modified_bessel_i0')
```

Comment (peterbell10) [reply torch/_inductor/lowering.py:599]: Having `register_pointwise_numeric` accept other keyword args SGTM.

**Review by peterbell10 — APPROVED** [review]

**Comment by isuruf** [comment]
Is this PR the cause of #119487, #119488, #119489 ?

**Comment by peterbell10** [comment]
Looks like a coincidence to me. There's no reason these changes should effect `make_fx` and the failures actually started at 4:23 pm whereas this wasn't merged until 6:42 pm.  I also found this old issue that looks very similar #107114. I do wonder why they're all modified bessel functions though, that just seems odd.

**Pull request closed without merging**